File: solutions/Majority-Element/Majority-Element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function (nums) {
  let ans = nums[0];
  let cnt = 1;
  const length = nums.length;
  for (let i = 1; i < length; i++) {
    if (cnt === 0) {
      ans = nums[i];
    }
    cnt = nums[i] === ans ? cnt + 1 : cnt - 1;
  }
  return ans;
};

File: solutions/Majority-Element/Majority-Element.cpp
// 2019-07-28
// O(n)
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int cnt = 0;
        int ans = 0;
        for(int i=0;i<nums.size();i++)
        {
            if(cnt == 0)
            {
                ans = nums[i];
            }
            cnt += nums[i] == ans ? 1 : -1;
        }
        return ans;
    }
};